micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationContextAssert.java

```diff
@@ -197,6 +200,24 @@ else if (!extraKeys.isEmpty()) {
         return (SELF) this;
     }
 
+    /**
+     * Verifies that the Observation key-value keys are a subset of the given set of keys.
+     */
+    public SELF hasSubsetOfKeys(String... keys) {
```

Comment:
Added this to the asserts because it may be useful to others in other places, but also I needed it for the verification with the `ObservationDocumentation` in the TCK.